### ISSUE TITLE
Learning rate finder for the trainer 

### ISSUE THREAD

**Issue by franz101** [huggingface/transformers#16013]
Hi,

is it possible to integrate a learning rate finder with the trainer api?
Something like this:
https://github.com/davidtvs/pytorch-lr-finder

It's probably has been done before, but couldn't find it yet.

**Comment by sgugger** [comment]
The LR finder does not give reliable results for Transformers models (usually indicates one in the 1e-3 when the actual best value is in the 1e-5 range). That's why we don't have support for it.
